db/view/view.cc

```diff
@@ -1203,8 +1204,9 @@ future<> mutate_MV(
         service::allow_hints allow_hints,
         wait_for_all_updates wait_for_all)
 {
-    auto fs = std::make_unique<std::vector<future<>>>();
-    fs->reserve(view_updates.size());
+    static constexpr size_t max_concurrent_updates = 128;
+    std::vector<future<>> fs;
+    fs.reserve(std::min(view_updates.size(), max_concurrent_updates));
     for (frozen_mutation_and_schema& mut : view_updates) {
```

Comment:
I'll try to apply something like that, but in case of this function perhaps a regular semaphore will be enough to keep the number of active continuations below 128.

Reply:
max_concurrent_for_each() is exactly that mix of a semaphore and parallel_for_each.